test/ServiceTest.cpp

```diff
@@ -408,6 +408,31 @@ TEST_F(ServiceTest, computeResult) {
         siblingTree};
     EXPECT_NO_THROW(serviceOperation6.computeResultOnlyForTesting());
     RuntimeParameters().set<"service-max-value-rows">(maxValueRowsDefault);
+
+    // Check 7: Lazy computation
+    Service lazyService{
+        testQec, parsedServiceClause,
+        getResultFunctionFactory(
+            expectedUrl, expectedSparqlQuery,
+            genJsonResult({"x", "y"},
+                          {{"bla", "bli"}, {"blu", "bla"}, {"bli", "blu"}}),
+            boost::beast::http::status::ok, "application/sparql-results+json")};
+
+    auto lazyResult = lazyService.computeResultOnlyForTesting(true);
+    // Consume the idTable-generator to actually compute the result.
+    for ([[maybe_unused]] auto& _ : lazyResult.idTables()) {
+    }
+
+    // Check 8: LazyJsonParser Error
+    Service service8{
+        testQec, parsedServiceClause,
+        getResultFunctionFactory(
+            expectedUrl, expectedSparqlQuery, std::string(1'000'000, '0'),
+            boost::beast::http::status::ok, "application/sparql-results+json")};
+
+    AD_EXPECT_THROW_WITH_MESSAGE(
+        service8.computeResultOnlyForTesting(),
```

Comment:
This is not lazy, am i right?
When being lazy, you should probably see an error during the consumation of the generator (at least for the `invalid Json `kind of error.